PLW0120 false positive for/with/break/else

* A minimal code snippet that reproduces the bug.

```python
from contextlib import suppress

for name in ["demo"]:
    with suppress(AttributeError, KeyError):
        thing = name
        break
else:
    thing = "other stuff"
```

* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.

```console
% ruff demo.py
demo.py:1:1: D100 Missing docstring in public module
demo.py:7:1: PLW0120 `else` clause on loop without a `break` statement; remove the `else` and de-indent all the code inside it
Found 2 errors.
```

* The current Ruff settings (any relevant sections from your `pyproject.toml`).

Just enabling a bunch of letters, plus targeting Python 3.7.

* The current Ruff version (`ruff --version`).

```console
% ruff --version
ruff 0.0.247
```

--- a/crates/ruff/resources/test/fixtures/pylint/useless_else_on_loop.py
+++ b/crates/ruff/resources/test/fixtures/pylint/useless_else_on_loop.py
@@ -113,3 +113,14 @@ def test_break_in_if_orelse():
     else:
         return True
     return False
+
+
+def test_break_in_with():
+    """no false positive for break in with"""
+    for name in ["demo"]:
+        with open(__file__) as f:
+            if name in f.read():
+                break
+    else:
+        return True
+    return False

--- a/crates/ruff/src/rules/pylint/rules/useless_else_on_loop.rs
+++ b/crates/ruff/src/rules/pylint/rules/useless_else_on_loop.rs
@@ -22,6 +22,7 @@ impl Violation for UselessElseOnLoop {
 fn loop_exits_early(body: &[Stmt]) -> bool {
     body.iter().any(|stmt| match &stmt.node {
         StmtKind::If { body, orelse, .. } => loop_exits_early(body) || loop_exits_early(orelse),
+        StmtKind::With { body, .. } | StmtKind::AsyncWith { body, .. } => loop_exits_early(body),
         StmtKind::Try {
             body,
             handlers,

